src/main/frontend/handler/repo.cljs

```diff
@@ -473,27 +473,21 @@
              (state/set-db-restoring! false)))
     (js/setTimeout setup-local-repo-if-not-exists! 100)))
 
-(defn periodically-pull
-  [repo-url pull-now?]
-  (spec/validate :repos/url repo-url)
-  (p/let [token (helper/get-github-token repo-url)]
-    (when token
-      (when pull-now? (pull repo-url nil))
-      (js/setInterval #(pull repo-url nil)
-                      (* (config/git-pull-secs) 1000)))))
-
-(defn periodically-push-tasks
-  [repo-url]
-  (js/setInterval #(push-if-auto-enabled! repo-url)
+(defn periodically-pull-current-repo
+  []
+  (js/setInterval
+   (fn []
+     (p/let [repo-url (state/get-current-repo)
+             token (helper/get-github-token repo-url)]
```

Comment:
Calling `get-github-token` inside the task, is it correct?

Reply:
I don't quite get it, we need a token here to pull, right?

Reply:
> we need a token here to pull, right?

Yes. But the old logic is for each repo, only starts the `js/setInterval` task when the token can be acquired.

Reply:
Gotcha, I think it's correct to get the token in the interval